.github/workflows/pre-commit.yaml

```
@@ -0,0 +1,58 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow
+
+# Same name as the file for consistent identification
+name: pre-commit
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "*"
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  pre-commit:
+    name: Detecting code style issues
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install ALSA and Jack dependencies
+        run: |
+          sudo apt-get update && sudo apt-get install -y libasound2-dev libjack-jackd2-dev
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@master
```

Is this an improved version over `actions-rs/toolchain@v1` ? In that case we should probably also use it in `rust.yaml`

Also, maybe best to not depend on bleeding edge?
```suggestion
        uses: dtolnay/rust-toolchain@v1
```

---

Yes. actions-rs/toolchain@v1 is unmaintained and has issues. I have switched all my projects over.

---

I don't remember why I have chosen `master` over `v1`. It never caused any issues and it doesn't hurt unless the CI build breaks. Since we use the most recent Rust version for the CI builds we should better rely on the most recent tooling.

---

According to the README `@stable` is what we want: https://github.com/orottier/web-audio-api-rs/pull/290/commits/336f65b647196fc0bd392655cafb1675e23e7eef